### DISCUSSION TITLE
Show popup if regex not found?

### DISCUSSION THREAD

**Discussion by zohozer** [facelessuser/RegReplace#111] [Q&A]
I am trying to build a custom RegReplace rule that needs to work only in some specific files. More precisely, I created a rule "caption": "Reg Replace: Gcode r130" that I do need to be able to run ONLY in files that do have the following data into the first line of the text file:

`H DX=(2440.000) DY=(1220.000) DZ=(18.700) -EH C=0 T=16973824`

The data between parentheses are variables, so they can change from file to file.

My find regex looks like: `"find": " -EH C=0 T=16973824(.*)$",`
What I do need to put in the `"replace":` field to have a warning popup if the Find: part don't find anything that matches this rule?

If I do try to run the `Gcode r130` rule, I do need to get a popup that says something like: "Not a valid file for conversion" and to stop any further execution.

It is possible to create something like this?

I do create a custom RegReplace rule (called `Gcode r130`) to convert some Gcode files from a format to another format and I do need to be able to run-it ONLY if I load the right files.

**Comment by facelessuser** [comment]
Sorry, it took me so long to get to this. I don't currently have conditional rules to prevent a pattern from running.

You can add an issue for a feature request so that such a feature could be considered for the future.
